builtin/mainmenu/dlg_settings_advanced.lua

```diff
@@ -854,7 +859,27 @@ local function handle_change_setting_buttons(this, fields)
 	return false
 end
 
-local function create_settings_formspec(tabview, name, tabdata)
+local function filter_world_settings()
```

Comment:
Currently it only filters top-level catergories, aren't mod settings are all under the mods catergory? You are right it would be better to add this as an attribute to the settingtypes file.